File: solutions/fsharp/log-levels/1/LogLevels.fs
module LogLevels

let message (logLine: string): string = logLine[logLine.IndexOf(' ')+1..].Trim()

let logLevel(logLine: string): string = logLine[logLine.IndexOf('[')+1..logLine.IndexOf(']')-1].ToLower()

let reformat(logLine: string): string = $"{message logLine} ({logLevel logLine})"

File: solutions/csharp/log-levels/1/LogLevels.cs
using System;

static class LogLine
{
    public static string Message(string logLine) => logLine.Split(' ', 2)[1].Trim();

    public static string LogLevel(string logLine)
    {
        int firstBracket = logLine.IndexOf('[');
        int secondBracket = logLine.IndexOf(']');
        return logLine.Substring(firstBracket + 1, secondBracket - firstBracket - 1).ToLower();
    }

    public static string Reformat(string logLine) => $"{Message(logLine)} ({LogLevel(logLine)})";
}
